pylib/mozhginfo/mozhginfo/pushlog_client.py

```diff
@@ -0,0 +1,187 @@
+        start_id - 1,  # off by one to compensate for pushlog as it skips start_id
+        end_id,
+        version
+    )
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    pushes = req.json()["pushes"]
+    # pushes.keys() is a list of strings which we need to map to integers
+    # We use reverse in order to return list sorted from newest to oldest push id
+    for push_id in sorted(map(int, pushes.keys()), reverse=True):
+        # Querying by push ID is preferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the 12 char representation
+        revisions.append(pushes[str(push_id)]["changesets"][0])
+
+    return revisions
+
+
+def query_revisions_range_from_revision_before_and_after(repo_url, revision, before, after):
+    """
+    Get the start and end revisions based on the number of revisions before and after.
+
+    Raises PushlogError if pushlog data cannot be retrieved.
+    """
+    try:
+        push_info = query_revision_info(repo_url, revision)
+        pushid = int(push_info["pushid"])
+        start_id = pushid - before
+        end_id = pushid + after
+        revlist = query_pushid_range(repo_url, start_id, end_id)
+    except:
+        raise PushlogError('Unable to retrieve pushlog data. '
+                           'Please check repo_url %s and revision %s specified.'
+                           % (repo_url, revision))
+
+    return revlist
+
+
+def query_revision_info(repo_url, revision, full=False):
+    """
+    Return a dictionary with meta-data about a push including:
+
+        * changesets
+        * date
+        * user
+    """
+    url = "%s?changeset=%s&tipsonly=1" % (JSON_PUSHES % {"repo_url": repo_url}, revision)
+    if full:
+        url += "&full=1"
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    data = req.json()
+    assert len(data) == 1, "We should only have information about one push"
+    push_id, push_info = data.popitem()
+    push_info["pushid"] = push_id
+    if not full:
+        LOG.debug("Push info: %s" % str(push_info))
+    else:
+        LOG.debug("Requesting the info with full=1 can yield too much unnecessary output "
+                  "to debug anything properly")
+    return push_info
+
+
+def query_full_revision_info(repo_url, revision):
+    """
+    Return the full 40 chars revision number which correspond to the 12 chars
+    revision number we given, and throw an error out if found collision in there.
+
+    :param repo_url: The repo url of a repository e.g. mozilla-inbound
+    :param revision: push revision (40 chars)
+    :type revision: str
+    :return:
+    """
+    revision = query_revision_info(repo_url, revision)['changesets'][0]
+    if not revision:
+        raise PushlogError('Unable to retrieve pushlog info by repo_url: %s'
+                           ' and revision: %s. Please check repo_url and revision secified.'
```

Comment:
Since we know that this error can be caused by:
- The revision doesn't exist in the repo
- We don't have enough characters to determine the revision
We should inform our users of these possibilities.